python/tests/prototype/serial.py

```diff
@@ -0,0 +1,138 @@
+"""
+ * GTDynamics Copyright 2020, Georgia Tech Research Corporation,
+ * Atlanta, Georgia 30332-0415
+ * All Rights Reserved
+ * See LICENSE for the license information
+ *
+ * @file  serial.py
+ * @brief Ptototype Serial class for serial manipulators.
+ * @author Frank Dellaert
+"""
+
+from typing import Optional, Tuple
+
+import gtdynamics as gtd
+import numpy as np
+from gtsam import Pose3
+
+
+def compose(A: Tuple[Pose3, np.ndarray], B: Tuple[Pose3, np.ndarray]):
+    """Monoid operation for pose,Jacobian pairs."""
+    aTb, Jb = A
+    bTc, Jc = B
+    assert Jb.shape[0] == 6 and Jc.shape[0] == 6, f"{Jb.shape} and {Jc.shape}"
+
+    # Compose poses
+    aTc = aTb.compose(bTc)
+
+    # Check if one of specs is an offset:
+    if Jb.shape[1] == 0:
```

Comment:
Are these ifs really needed? 
If Jb or Jc are empty, they will just not add any new columns when stacking. 
I see that for the Jb if, you evade computing the adjointmap, but the Jc one doesn't solve much.

Reply:
Thanks, much simpler without checks and still works!